### PR TITLE
Define external tables, more dbt setup, first models

### PR DESCRIPTION
**Pull request by lauriemerrell**
## Description
*What is the purpose of your pull request?*

* Creates external tables to read parsed data into BigQuery
* Creates initial staging tables for GTFS-RT specifically
* Updates READMEs to be more specific to our project 
* Adds PR template

## Testing
*How have you tested your pull request?*

``` sql
SELECT * 
FROM `transit-data-analytics-demo.laurie_external_transit_data.gtfs_schedule__agency` 
WHERE dt = '2023-07-18' AND hour = '2023-07-18 00:00:00 UTC'
```

``` sql
SELECT * 
FROM `transit-data-analytics-demo.laurie_staging.stg_gtfs_rt__trip_updates` 
WHERE dt = '2023-07-18' AND hour = '2023-07-18 00:00:00 UTC'
ORDER BY _extract_ts
```

``` sql
SELECT * 
FROM `transit-data-analytics-demo.laurie_external_transit_data.gtfs_schedule__agency` 
WHERE dt = '2023-07-18' AND hour = '2023-07-18 00:00:00 UTC'
```

### CODE REVIEW

File: warehouse/macros/get_string_value_at_json_path.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{% macro get_string_value_at_json_path(column, json_relative_path) %}
+STRING(JSON_QUERY({{ column }}, '$.{{ json_relative_path }}'))
```
Comment (atvaccaro) [inline warehouse/macros/get_string_value_at_json_path.sql:2]: maybe the type conversion could be an argument with a default of `STRING`?

File: warehouse/macros/parsed_file_common_element_lookups.sql [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        TIMESTAMP({{ get_string_value_at_json_path('file', 'ts') }}) AS _extract_ts
+
+{% endmacro %}
+
+{% macro metadata_columns() %}
+    dt,
+    hour,
+    _url,
+    _name,
+    _extract_ts
```
Comment (atvaccaro) [inline warehouse/macros/parsed_file_common_element_lookups.sql:18]: nit: probably want a trailing comma for these

File: warehouse/macros/parsed_file_common_element_lookups.sql [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+-------------- GTFS RT --------------
+
+-- given the path to the `trip` element (something like `entity.vehicle`, does not include initial `$.`)
+-- parse a trip descriptor from JSONL
+{% macro read_gtfs_rt_trip_descriptor(parent_json_path) %}
+
+        {{ get_string_value_at_json_path('record', parent_json_path + '.trip.tripId') }} AS trip_id,
+        {{ get_string_value_at_json_path('record', parent_json_path + '.trip.routeId') }} AS trip_route_id,
+        {{ get_string_value_at_json_path('record', parent_json_path + '.trip.direectionId') }} AS trip_direction_id,
```
Comment (atvaccaro) [inline warehouse/macros/parsed_file_common_element_lookups.sql:29]: `direectionId`

**Review by atvaccaro — APPROVED** [review]
I think the direction typo is the only blocker

**Comment by atvaccaro** [comment]
this looks great, glad we've learned our lessons 🤣

File: warehouse/macros/parsed_file_common_element_lookups.sql (hunk 2 above)
Comment (lauriemerrell) [reply warehouse/macros/parsed_file_common_element_lookups.sql:18]: disagree, see the usage -- I think it's more natural to use a comma after the macro call if one is needed (two macros back to back with no comma look weird to me?)

**Pull request merged**